numpy/core/arrayprint.py

```diff
@@ -1236,7 +1238,7 @@ def dtype_is_implied(dtype):
     dtype = np.dtype(dtype)
     if _format_options['legacy'] == '1.13' and dtype.type == bool_:
         return False
-    return dtype.type in _typelessdata
+    return dtype.type in _typelessdata and dtype.names is None
```

Comment:
As we're adding comments, maybe here note that the latter part guards against union dtypes?